File: backend/app/models/column.rb
# frozen_string_literal: true

class Column < ApplicationRecord
  self.inheritance_column = :_type_disabled

  belongs_to :table
  belongs_to :foreign_key_table,
             class_name: 'Table',
             optional: true
  has_one :memo,
          dependent: :destroy,
          class_name: 'ColumnMemo'

  def self.name_like(keyword)
    return all if keyword.blank?

    where('columns.name LIKE ?', "%#{sanitize_sql_like(keyword)}%")
  end
end

File: frontend/types/column.ts
import { Table } from './table';
import { ColumnMemo } from './column_memo';

export type Column = {
  id: number;
  name: string;
  comment: string;
  type: string;
  nullable: boolean;
  primary_key: boolean;
  table: Table;
  foreign_key_table: Table;
  memo: ColumnMemo;
}
